presto-cache/src/main/java/com/facebook/presto/cache/LocalRangeCacheManager.java

```diff
@@ -261,17 +263,32 @@ private boolean write(FileReadRequest key, byte[] data, Path newFilePath)
             else {
                 // copy previous file's data to the new file
                 byte[] previousFileBytes = Files.readAllBytes(new File(previousCacheFile.getPath().toUri()).toPath());
-                Files.write((new File(newFilePath.toUri())).toPath(), previousFileBytes, CREATE_NEW);
+                File newFile = new File(newFilePath.toUri());
+                Files.write(newFile.toPath(), previousFileBytes, CREATE_NEW);
                 int previousFileLength = previousFileBytes.length;
                 long previousFileOffset = previousCacheFile.getOffset();
 
                 // remove the overlapping part and append the remaining cache data
-                byte[] remainingCacheFileBytes = new byte[toIntExact((key.getLength() + key.getOffset()) - (previousFileLength + previousFileOffset))];
-                System.arraycopy(data, toIntExact(previousFileLength + previousFileOffset - key.getOffset()), remainingCacheFileBytes, 0, remainingCacheFileBytes.length);
```

Comment:
`data` is usually small; so we can append it to `newFilePath` once. But can we avoid an extra copy from `data` to `remainingCacheFileBytes` and then to `newFilePath`? Feel free to make it more efficient thought the original code isn't

Reply:
this context regarding ```data``` size is useful :) Changed to using a ```RandomAccessFile``` instead which should avoid the extra array copy. What do you think?